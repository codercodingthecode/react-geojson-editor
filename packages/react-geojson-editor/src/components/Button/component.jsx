import React from 'react'; // eslint-disable-line import/no-extraneous-dependencies
import PropTypes from 'prop-types'; // eslint-disable-line import/no-extraneous-dependencies

import ButtonStyles from './styles.scss';

export const Button = ({ children, ...props }) => (
  <button
    className={ButtonStyles.button}
    {...props}
  >
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.string,
};

Button.defaultProps = {
  children: 'Button',
};

export default Button;
