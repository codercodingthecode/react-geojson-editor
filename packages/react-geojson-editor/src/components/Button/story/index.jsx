import React from 'react'; // eslint-disable-line import/no-extraneous-dependencies

import Dropdown from '../component';

/**
 * Storybook for Dropdown
 */
export default (storiesOf, module, ContainerDecorator) => {
  storiesOf('Dropdown', module)
    .addDecorator(ContainerDecorator)
    .add('with required props', () =>
      (
        <Dropdown name="Customize View" >
          <div style={{ padding: 0, height: 250 }}> Child Component </div>
        </Dropdown>
      ))
    .add('with right positioning prop', () =>
      (
        <Dropdown name="Customize View" positionRight>
          <div style={{ padding: 20, height: 250 }}> Child Component </div>
        </Dropdown>
      ))
    .add('with default props', () => <Dropdown />);
};
