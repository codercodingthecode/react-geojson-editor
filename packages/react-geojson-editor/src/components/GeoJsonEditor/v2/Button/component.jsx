import React, { Component, Children } from 'react';

import ButtonStyles from './styles.scss';

export const Button = ({ children, ...props }) => (
  <button
    className={ButtonStyles.button}
    {...props}
  >{children}
  </button>
);

export default Button;
