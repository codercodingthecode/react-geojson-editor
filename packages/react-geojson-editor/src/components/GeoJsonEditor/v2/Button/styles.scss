.button {
    outline: none;
    cursor: pointer;
    display: inline-block;
    width: 100%;
    // height: 40px;
    // height: inherit;
    background: #FFFFFF;
    // border: 1px solid #AAA9AA;
    border: none;
    border-bottom: 1px solid #d4d6d8;
    // box-shadow: rgba(0, 0, 0, 0.3) 0px 2px 6px;
    border-radius: 0;
    // font-family: 'Noto Sans', sans-serif;
    font-size: 13px;
    line-height: 17px;
    color: #333333;
    text-align: left;
    padding: 10px;
    position: relative;
    top: 1px;

    &:disabled {
        opacity: 0.5;
    }
}