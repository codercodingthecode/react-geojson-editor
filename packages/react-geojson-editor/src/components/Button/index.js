import component from './component';

export { Button } from './component';
// export { default as DropdownStory } from './story';

export default component;
