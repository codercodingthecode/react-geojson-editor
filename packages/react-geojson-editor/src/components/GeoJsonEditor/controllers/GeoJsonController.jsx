import React from 'react';
import PropTypes from 'prop-types';
import { GeoJsonEditorMode } from '../../../utils';
import { Dropdown } from '../../Dropdown';
import { Button } from '../../Button';
import { PolygonDataContext } from '../google-map'; // @TODO
import Controls from './Controls';

/**
 * - show ui for enabling polygon editing capability
 * - dispatch actions for
 *      - persting new changes
 *      - remove new changes
 *      - toggle edit mode
 * @param {*} props
 */
export const GeoJsonController = ({
  mode, onToggleMode,
}) => (
  <Controls title="Managing geojson changes">
    {mode !== GeoJsonEditorMode.VIEW ?
      <Dropdown name="Changes" openAtStart style={{ padding: 0 }} containerStyle={{ width: 120 }} >
        <PolygonDataContext.Consumer>
          {({ discardChanges, persistChanges }) =>
                        (
                          <div style={{ padding: 0 }} title="GeoJsonControls">
                            <Button
                              onClick={(ev) => {
                                        persistChanges(
                                            ev,
                                            () => onToggleMode(GeoJsonEditorMode.VIEW),
                                        );
                                    }}
                              title="Save changes"
                            >
                                    Save
                            </Button>
                            <Button
                              onClick={ev => discardChanges(
                                        ev,
                                        () => onToggleMode(GeoJsonEditorMode.VIEW),
                                    )
                                    }
                              title="Discard changes"
                            >
                                    Cancel
                            </Button>
                          </div>
                        )
                    }
        </PolygonDataContext.Consumer>
      </Dropdown> :
      <Button
        onClick={() => onToggleMode(GeoJsonEditorMode.EDIT)}
        style={{
            width: 120,
            boxShadow: 'rgba(0, 0, 0, 0.3) 0px 2px 6px',
            borderRadius: 2,
        }}
        title="Turn on Edit mode"
      >
                Create / Edit
      </Button>
        }
  </Controls>
);
export default GeoJsonController;

GeoJsonController.propTypes = {
  mode: PropTypes.string,
  onToggleMode: PropTypes.func,
};

GeoJsonController.defaultProps = {
  mode: GeoJsonEditorMode.VIEW,
  onToggleMode: () => { },
};
