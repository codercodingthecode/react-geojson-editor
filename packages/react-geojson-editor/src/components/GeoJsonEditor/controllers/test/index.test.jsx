import React from 'react';

import { shallow } from 'enzyme';
import { GeoJsonController } from '../GeoJsonController';
import { Button } from '../../../Button';

describe('Button', () => {
  it('has not changed', () => {
    const wrapper = shallow(<GeoJsonController />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders', () => {
    const wrapper = shallow(<GeoJsonController />);
    expect(wrapper.find(Button)).toHaveLength(1);
  });
});
