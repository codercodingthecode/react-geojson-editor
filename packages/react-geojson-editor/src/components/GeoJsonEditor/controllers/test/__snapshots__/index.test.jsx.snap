// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has not changed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    mode="VIEW"
    onToggleMode={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Button
        onClick={[Function]}
        style={
          Object {
            "borderRadius": 2,
            "boxShadow": "rgba(0, 0, 0, 0.3) 0px 2px 6px",
            "width": 120,
          }
        }
        title="Turn on Edit mode"
      >
        Create / Edit
      </Button>,
      "title": "Managing geojson changes",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Create / Edit",
        "onClick": [Function],
        "style": Object {
          "borderRadius": 2,
          "boxShadow": "rgba(0, 0, 0, 0.3) 0px 2px 6px",
          "width": 120,
        },
        "title": "Turn on Edit mode",
      },
      "ref": null,
      "rendered": "Create / Edit",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Button
          onClick={[Function]}
          style={
            Object {
              "borderRadius": 2,
              "boxShadow": "rgba(0, 0, 0, 0.3) 0px 2px 6px",
              "width": 120,
            }
          }
          title="Turn on Edit mode"
        >
          Create / Edit
        </Button>,
        "title": "Managing geojson changes",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Create / Edit",
          "onClick": [Function],
          "style": Object {
            "borderRadius": 2,
            "boxShadow": "rgba(0, 0, 0, 0.3) 0px 2px 6px",
            "width": 120,
          },
          "title": "Turn on Edit mode",
        },
        "ref": null,
        "rendered": "Create / Edit",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
